### PR TITLE
Locale as in system

### LINKED ISSUE #224: Locale as in system

### PR DESCRIPTION
**Pull request by nyet-ty**
Resolve #224

### CODE REVIEW

File: src/middleware.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { NextResponse } from 'next/server';
+import type { NextRequest } from 'next/server';
+
+export const languages = ['en', 'ru'];
```
Comment (9teen90nine) [inline src/middleware.ts:4]: https://github.com/taskany-inc/hire/blob/main/src/utils/getLang.ts#L1

File: src/middleware.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    ) {
+        return NextResponse.next();
+    }
+    const acceptLanguage = request.headers.get('Accept-Language');
+
+    if (
+        acceptLanguage &&
+        languages.includes(acceptLanguage.slice(0, 2)) &&
+        !request.url.includes(`/${acceptLanguage.slice(0, 2)}/`) &&
+        request.nextUrl.locale !== acceptLanguage.slice(0, 2)
```
Comment (9teen90nine) [inline src/middleware.ts:21]: `acceptLanguage.slice(0, 2)` should be a variable

File: src/middleware.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import type { NextRequest } from 'next/server';
+
+export const languages = ['en', 'ru'];
+
+export function middleware(request: NextRequest) {
+    if (
+        request.nextUrl.pathname.startsWith('/api') ||
+        request.nextUrl.pathname.startsWith('/_next') ||
+        request.nextUrl.pathname.startsWith('/favicon') ||
+        request.nextUrl.pathname === '/'
```
Comment (9teen90nine) [inline src/middleware.ts:11]: 1. Why not redirect root? You should actually. Also, it's better to disable next locale detection.
2. What happens if a route already includes a locale other than current?

File: src/middleware.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        request.nextUrl.pathname.startsWith('/api') ||
+        request.nextUrl.pathname.startsWith('/_next') ||
+        request.nextUrl.pathname.startsWith('/favicon')
+    ) {
+        return NextResponse.next();
+    }
+
+    const localeFromHeader = acceptLanguage.slice(0, 2);
+
+    if (languages.some((language) => request.url.includes(`/${language}/`)) || localeFromHeader === defaultLocale) {
```
Comment (tsumo) [inline src/middleware.ts:20]: Not `includes`, but `startsWith`

File: src/middleware.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+        return NextResponse.next();
+    }
+
+    if (languages.some((language) => localeFromHeader === language) && defaultLocale !== localeFromHeader) {
+        return NextResponse.redirect(`${request.nextUrl.origin}/${localeFromHeader}${request.nextUrl.pathname}`);
+    }
+}
```
Comment (tsumo) [inline src/middleware.ts:27]: No default return? What if all conditions were skipped?

File: src/middleware.ts (hunk 4 above)
Comment (nyet-ty) [reply src/middleware.ts:20]: No, 'request.url' starts with 'https...'

File: src/middleware.ts (hunk 5 above)
Comment (nyet-ty) [reply src/middleware.ts:27]: My bad

**Review by tsumo — APPROVED** [review]

**Review by 9teen90nine — APPROVED** [review]

**Pull request merged**